### PR TITLE
Adding executable to run surveyor from the command line.

### LINKED ISSUE #2128: Sufia 6.x_stable Survey
I would like to be able to survey my Sufia 6.x repository prior to migration, so that I can verify that every object has been moved.  Part of user story #2127 

I would like to create a executable I can run that creates a database table with survey information.

This will be done when I can run sufia_survey.exe in my sufia 6.x  application and I produce a database table that contains a list of my object ids and types [Collections & Files].

My executable should take the following options:
  --limit= [all (default), # of objects,]   # limit the number of items surveyed
  --id=  a list of Id (id1,id2)                 # only survey the given ids

### PR DESCRIPTION
**Pull request by cam156**
Fixes #2128

Adds an executable to run the surveyor from the command line.  In addition this PR fixes a "bug" in the surveyor, to update an Survey::Item instead of always creating one.  That way if you run the sufia_survey 3 times you only get one item per Fedora object instead of 3.

@projecthydra/sufia-code-reviewers

### CODE REVIEW

File: exe/sufia_survey [hunk 1]
```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env ruby
+require 'rails'
+
+require ::File.join(File.dirname(Bundler.default_gemfile), 'config/environment')
+
+options = {}
+
+subtext = <<HELP
+Commonly used command are:
```
Comment (mjgiarlo) [inline exe/sufia_survey]: `s/command/commands/`

File: exe/sufia_survey [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+   clean :     cleans all Sufia::Migration::Survey::Items from the database
+See 'sufia_survey COMMAND --help' for more information on a specific command.
+HELP
+
+args = OptionParser.new do |opts|
+  opts.banner = "Survey model objects in fedora and store it in an ActiveRecord table. Usage: sufia_survey [options]"
```
Comment (mjgiarlo) [inline exe/sufia_survey]: Store what? The results of the survey? Maybe reword as `s/store it/store survey results/`?

File: exe/sufia_survey [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+  opts.on("-v", "--[no-]verbose", "Run verbosely") do |v|
```
Comment (mjgiarlo) [inline exe/sufia_survey]: So `--no-verbose` and `--verbose` both trigger verbose output? Maybe I misunderstand what OptionParser is doing here.

File: exe/sufia_survey [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+    options[:limit] = v
+  end
+
+  opts.on("--ids ID_LIST", Array, "Survey only the objects in the list of IDs",
+          "For example to survey a Collection with the id abc123 and a GenericFile with an id def456:",
+          "  --ids abc123,def456") do |v|
+    options[:ids] = v
+  end
+
+  opts.on("--models MODEL_LIST", Array, "Specify the models to be surveyed (defaults to GenericFile,Collection)") do |v|
```
Comment (mjgiarlo) [inline exe/sufia_survey]: Looks like lines 31-33 are identical to lines 35-37.

File: exe/sufia_survey [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+  end
+  args.parse!
+rescue => error
+  $stderr.puts "ERROR: #{error}\n"
+  $stderr.puts args.help
+  exit 1
+end
+
+if command == 'clean'
+  $stderr.puts "cleaning Sufia::Migration::Survey::Item"
```
Comment (mjgiarlo) [inline exe/sufia_survey]: `s/Item/Items/` ?

Comment (mjgiarlo) [inline exe/sufia_survey]: Should this go to `$stdout` instead of `$stderr`?

File: exe/sufia_survey [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+
+$stdout.puts "running with #{options}" if verbose
+id_service = Sufia::Migration::Survey::FedoraIdService.new
+
+if options[:models]
+  model_options = options.delete(:models)
+  $stdout.puts "processing models #{model_options}" if verbose
+  model_options.each do |model_str|
+    model_class = Object.const_get(model_str)
+    id_service.register_model(model_class) unless id_service.model_registry.keys.include?(model_class)
```
Comment (mjgiarlo) [inline exe/sufia_survey]: I might like to see the ID service have a method so you could simplify this to `id_service.registered?(model_class)` but I don't care *too* much about that.

File: exe/sufia_survey (hunk 3 above)
Comment (cam156) [inline exe/sufia_survey]: @ mjgiarlo the [no] is essentially a not, so no-verbose forces not verbose.  I actually copied this out of solr-wrapper so that the executables would be consistent across hydra.  I'll change the wording to include [not].

File: exe/sufia_survey (hunk 5 above)
Comment (cam156) [inline exe/sufia_survey]: @mjgiarlo good 👀 .. copy pasta...
